### PR TITLE
[xhamster] Added support for mobile site; fixed thumbnails and tests

### PR DESCRIPTION
**Pull request by windom**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Added support for the mobile version of XHamster, by effectively redirecting to the desktop version. _VALID_URL had to be particularized a bit so that the regexps of XHamsterIE and XHamsterMobileIE don't overlap (`https?://(?:.+?\.)` was changed into `https?://(?:(?:www|[a-z]{2})\.)?`).
Fixed thumbnail extraction of XHamsterIE.
Fixed the test of XHamsterEmbedIE.

### CODE REVIEW

File: youtube_dl/extractor/xhamster.py [hunk 1]
```diff
@@ -231,3 +235,30 @@ def _real_extract(self, url):
             video_url = dict_get(vars, ('downloadLink', 'homepageLink', 'commentsLink', 'shareUrl'))
 
         return self.url_result(video_url, 'XHamster')
+
+
+class XHamsterMobileIE(XHamsterBase):
+    _VALID_URL = XHamsterBase._VALID_URL_TEMPLATE % 'm\.'
```
Comment (dstftw) [inline youtube_dl/extractor/xhamster.py]: Just modify the original extractor.

Comment (windom) [reply youtube_dl/extractor/xhamster.py]: Fair enough, done.

File: youtube_dl/extractor/xhamster.py [hunk 2]
```diff
@@ -75,6 +75,26 @@ class XHamsterIE(InfoExtractor):
         'params': {
             'skip_download': True,
         },
+    }, {
+        # mobile site
+        'url': 'https://m.xhamster.com/videos/cute-teen-jacqueline-solo-masturbation-8559111',
```
Comment (dstftw) [inline youtube_dl/extractor/xhamster.py]: Only matching.

Comment (windom) [reply youtube_dl/extractor/xhamster.py]: Changed.

**Pull request merged**